### ISSUE TITLE
Tiles outside grid's restricted_extent are being cached

### ISSUE THREAD

**Issue by fracaron** [mapserver/mapcache#194]
It seems tiles are being cached even if they are outside the grid's restricted_extent.

```
    <tileset name="mirador_mvt">
        <source>mirador_wms_mvt</source>
        <grid restricted_extent="-8150000 5580000 -7600000 6100000" minzoom="6" maxzoom="15">GoogleMapsCompatible</grid>
        <cache>disk</cache>
        <format>MVT</format>
    </tileset>
```

As a quick test, I have used the mapcache_seed to delete the cached tile with the same extent parameter and it deleted all tiles except those outside the restricted_extent.

**Comment by jbo-ads** [comment]
I just found an undocumented attribute which could explain and solve this issue.

By default, a tolerance of five tiles is added to the restricted extent, which means that the actual restricted extent is larger than the specified one with `restricted_extent` attribute.

The undocumented attribute is named `tolerance`. It is used as illustrated here:

    <grid restricted_extent="-8150000 5580000 -7600000 6100000" tolerance="0" minzoom="6" maxzoom="15">GoogleMapsCompatible</grid>

**Comment by jbo-ads** [comment]
Merged pull request [#296](https://github.com/mapserver/docs/pull/296) on mapserver/docs adds documentation to that `tolerance` attribute, see [Configuration File - Tileset](https://mapserver.org/mapcache/config.html#tileset).

Does it fully address this issue?
